tests/rptest/services/redpanda.py

```diff
@@ -276,6 +276,7 @@ class SISettings:
 
     GLOBAL_ABS_STORAGE_ACCOUNT = "abs_storage_account"
     GLOBAL_ABS_SHARED_KEY = "abs_shared_key"
+    GLOBAL_CLOUD_PROVIDER = "cloud_provider"
```

Comment:
Would it make sense to use this global flag to determine across all cloud providers when running in CDT (i.e. "gcp", "aws", "azure" would be supported values). It's currently quite complicated: we start with `CloudStorageType.AUTO` and if running on dedicated nodes (i.e. in CDT), we infer the cloud from the global variables. We would still need to keep the logic that peeks at the `injecte_args` in the context for dockerised runs (they don't have the globals IIRC), but it might simplify things a bit.

Reply:
Totally make sense.
I have raised an issue to make this change. https://github.com/redpanda-data/redpanda/issues/9206